controllers/metallb_controller.go

```diff
@@ -135,18 +135,23 @@ func (r *MetalLBReconciler) reconcileResource(ctx context.Context, req ctrl.Requ
 }
 
 func (r *MetalLBReconciler) SetupWithManager(mgr ctrl.Manager, bgpType, enableWebhook string) error {
-	if bgpType == "" {
-		bgpType = bgpNative
-	}
-	if bgpType != bgpNative && bgpType != bgpFrr {
-		return fmt.Errorf("unsupported BGP implementation type: %s", bgpType)
-	}
-	if enableWebhook == "true" {
-		ManifestPath = fmt.Sprintf("%s/%s-with-webhooks", ManifestPath, bgpType)
+	logger := r.Log.WithName("SetupWithManager")
+	if r.PlatformInfo.IsOpenShift() {
+		logger.Info("this is openshift cluster. choosing its frr manifest")
+		ManifestPath = fmt.Sprintf("%s/openshift", ManifestPath)
```

Comment:
I think makes sense to fail if webhook is disabled, as it's always be enabled in this configuration